x-pack/test/saved_object_api_integration/security_only/apis/find.ts

```diff
@@ -37,19 +38,19 @@ export default function({ getService }: TestInvoker) {
           response: createExpectRbacForbidden('globaltype'),
         },
         unknownType: {
-          description: 'forbidden login and find wigwags message',
-          statusCode: 403,
-          response: createExpectRbacForbidden('wigwags'),
+          description: 'bad request, type must be valid',
+          statusCode: 400,
```

Comment:
@kobelb this is a vector for information disclosure, but I'm not sure it's an issue. Using this, an underprivileged user would be able to enumerate the known types by inspecting the response codes (400 vs 403). I know at one point we had made an intentional decision to return consistent response codes for these types of scenarios. Is that still worthwhile, or are we ok with this? IMO this isn't a big deal, but I'd like your thoughts here.

Reply:
The good 'ol information disclosure... The main thing that we end up disclosing here is whether or not a specific plugin is installed, which could be used to detect specific instances that have a vulnerable plugin installed... To get around this, we could potentially remove the validation from the routes themselves and rely on the errors from the SavedObjectClient, which would get us consistent error messages.

How would you feel about doing the type based validation all within the SavedObjectsClient/Repository themselves @njd5475 ?

Reply:
Ok we should be covered now, that the Joi validation errors are no longer happening. And hopefully the proper permissions responses are being returned.